avocado/core/runner.py

```diff
@@ -395,8 +394,12 @@ def sigtstp_handler(signum, frame):     # pylint: disable=W0613
             self.job.log.debug('')
 
         self.result.check_test(test_state)
-        if not status.mapping[test_state['status']]:
-            failures.append(test_state['name'])
+        if test_state['fail_class'] == 'TestTimeoutSkip':
+            # The test was skipped but the job was interrupted due
+            # to the timeout.
+            summary.append('INTERRUPTED')
+        else:
+            summary.append(test_state['status'])
```

Comment:
Hmm, I had something else in mind. Instead of returning list of results I wanted to just report flags of what was wrong.

Something like:

```
summary = set()

if test_state['fail_class'] == 'TestTimeoutSkip':
    summary.add('INTERRUPTED')
elif test_state['fail_class']:
    summary.add('FAIL')
```
and then only check what flags were set in the job runner and return the most appropriate return code (later all of them). The benefit is speed and memory utilization.

_Note: On the other hand returning all test statuses could be useful to multi-test RFC. Anyway it's only an internal API so we can always change it._

Anyway I can live with both if they are consistent (instead of adding 'INTERRUPTED' it should store test statuses for all of them and the job should check if `fail_class` is `TestTimeoutSkip`.